statsd/sender.go

```diff
@@ -5,23 +5,21 @@ import (
 	"sync/atomic"
 )
 
-// SenderMetrics contains metrics about the health of the sender
-type SenderMetrics struct {
-	TotalSentBytes                uint64
-	TotalSentPayloads             uint64
-	TotalDroppedPayloads          uint64
-	TotalDroppedBytes             uint64
-	TotalDroppedPayloadsQueueFull uint64
-	TotalDroppedBytesQueueFull    uint64
-	TotalDroppedPayloadsWriter    uint64
-	TotalDroppedBytesWriter       uint64
+// senderTelemetry contains telemetry about the health of the sender
+type senderTelemetry struct {
+	totalPayloadsSent             uint32
```

Comment:
Why not `uint64`? Now that counters are forever increasing there's an even greater likelihood of overflow (`uint32` max value is 4294967294, which may not be large enough for long-running applications), we could use the additional bits. Am I missing something?

Reply:
Ah OK, I think I see what's going on, we send the delta between flushes as a count so `uint32` should be enough to accomodate for that.

Reply:
also, `count` are sent as float64, so a `uint64` would not fit and we need a type working with atomic. That why I fall back on uint32.